### DISCUSSION TITLE
Multiple character rotation issues

### DISCUSSION THREAD

**Discussion by tlsduf** [Sixze/ALS-Refactored#562] [Q&A]
1. I want to maintain the character's rotation when moving forward towards a wall. Currently, the character rotates to the side when moving towards a wall...

2. In the air, the character rotates to face the direction the camera is looking at. I want to stop this behavior.

3. When using 
SetActorRotation((-1 * GetActorForwardVector()).Rotation());
the character is momentarily set to the opposite direction and then returns to the original direction. I want to fix this.**

**Comment by ameaninglessname** [comment]
For 1 and 2, check out settings in the `CS_Als_Default`:
![image](https://github.com/user-attachments/assets/a2687bd3-1c47-49f8-8e33-d1f51b9cd0a0)

For 3, that should be caused by `AAlsCharacter::RefreshGroundedRotation`, looks like you need a custom rotation mode. (search `SetActorRotation` in `AlsCharacter.cpp` for more details)

**Reply by tlsduf** [reply]
Thank you for your response! For issue 1, it’s not perfect, but the character's rotation has been mitigated. Issue 2 is completely resolved. As for issue 3, I’m not sure why, but after tweaking a few things, it’s now fixed!

**Reply by tlsduf** [reply]
However, when the character is near a wall, the issue of the character rotating is still not fixed. Is there a solution for this?

**Reply by ameaninglessname** [reply]
> However, when the character is near a wall, the issue of the character rotating is still not fixed. Is there a solution for this?

What's your `rotation mode`? Could you upload a video to elaborate it?

**Reply by tlsduf** [reply]
my rotation mode is VelocityDirection i'm not using ViewDirection because my game need it.

here is youtube : 
https://youtu.be/zYTeGEulcVc

**Reply by ameaninglessname** [reply]
Could you upload video here, I don't want to login YT.

In `VelocityDirection` mode, actor rotation is oriented to "movement" by default, which make sense.
So, IMO, you should find a way to not change velocity(such as skipping `SlideAlongSurface` in `MovementComponent` which sounds like a bad idea😂) when nearing a wall.

**Reply by tlsduf** [reply]
When I move close to the wall and face it directly, the character rotates to the side.

**Reply by tlsduf** [reply]
https://github.com/user-attachments/assets/934579f4-8555-4416-86a5-fc1e9c8e9355

**Reply by ameaninglessname** [reply]
> 2024-10-04.12-15-00.mp4

You could try to override `UCharacterMovementComponent::SlideAlongSurface` to do nothing.
It's called from `UCharacterMovementComponent::MoveAlongFloor`.

Or consider overrding `UMovementComponent::ComputeSlideVector`

**Reply by tlsduf** [reply]
oh thanks! i'll try this.
